Armadillo/src/main/java/com/scribd/armadillo/Reducer.kt

```diff
@@ -261,6 +271,42 @@ internal object Reducer {
                         ))
                         .apply { debugState = newDebug }
             }
+            is OpeningLicenseAction -> {
+                val oldDrm = oldState.drmPlaybackState
+                return oldState
+                    .copy(drmPlaybackState = DrmState.LicenseOpening(action.type ?: oldDrm.drmType))
+            }
+            is LicenseAcquiredAction -> {
+                val oldDrm = oldState.drmPlaybackState
+                return oldState
+                    .copy(drmPlaybackState = DrmState.LicenseAcquired(action.type, oldDrm.expireMillis))
+            }
+            is LicenseExpiredAction -> {
+                val oldDrm = oldState.drmPlaybackState
+                return oldState
+                    .copy(drmPlaybackState = DrmState.LicenseExpired(oldDrm.drmType, oldDrm.expireMillis))
+            }
+            is LicenseExpirationDetermined -> {
+                val oldDrm = oldState.drmPlaybackState
+                return oldState
+                    .copy(drmPlaybackState = DrmState.LicenseUsable(oldDrm.drmType, action.expirationMilliseconds))
+            }
+            is LicenseReleasedAction -> {
+                val oldDrm = oldState.drmPlaybackState
+                return oldState
+                    .copy(drmPlaybackState = DrmState.LicenseReleased(oldDrm.drmType,oldDrm.expireMillis, oldDrm.isSessionValid))
```

Comment:
What's the difference between `LicenseReleased` and `NoDrm`?

Reply:
The content is still loaded and the player hasn't reset for new content yet.    Playback won't happen anymore, but old content states are still present.